### PR TITLE
Search by stop place code & Skip common words

### PR DESCRIPTION
**Pull request by mhicauber**
Search stop places by prefix

Matching  https://github.com/entur/abzu/pull/38 front end functionnality

I'm sorry, there's also in this branch another small functionnality allowing to skip some common words in stop places search (see `stopPlaces.search.commonWordsToIgnore` property). We mixed up some branches.

### CODE REVIEW

File: src/main/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilder.java [hunk 1]
```diff
@@ -115,6 +116,24 @@
...
+    private static final String SQL_SEARCH_BY_CODE = "s.id in (select id from stop_place sp" +
+            " join stop_place_key_values spkv on sp.id = spkv.stop_place_id " +
+            " join value_items vi on vi.value_id = spkv.key_values_id" +
+            " where substring(vi.items, 4, 1) = ':' and lower(substring(vi.items, 1, 3)) = :dataProducerCode)";
+
+
+    /**
+     * Configure some common words to be skipped during stop place search by name.
+     */
+    @Value("#{'${stopPlaces.search.commonWordsToIgnore}'.split(',')}")
```
Comment (csolem) [inline src/main/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilder.java]: I think it should be possible to skip the `.split` part of the expression? 
```
@Value("${stopPlaces.search.commonWordsToIgnore}")
private List<String> commonWordsToIgnore;
```

File: src/main/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilder.java [hunk 2]
```diff
@@ -115,6 +116,24 @@
...
+    Set<String> commonWordsToIgnore = new LinkedHashSet<>();
```
Comment (csolem) [inline src/main/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilder.java]: private

File: src/main/java/org/rutebanken/tiamat/rest/graphql/StopPlaceRegisterGraphQLSchema.java [hunk 3]
```diff
@@ -553,6 +553,12 @@ private GraphQLObjectType createAddressablePlaceObjectType(List<GraphQLFieldDefi
                 .defaultValue(Boolean.FALSE)
                 .description("If set to true, only stop places with valid tags are returned. If false, filter does not apply.")
                 .build());
+        arguments.add(GraphQLArgument.newArgument()
+                .name(SEARCH_WITH_CODE)
+                .type(GraphQLString)
+                .defaultValue(null)
+                .description("Search with code")
```
Comment (csolem) [inline src/main/java/org/rutebanken/tiamat/rest/graphql/StopPlaceRegisterGraphQLSchema.java]: Specify what code this is. Data producer prefix code for instance.

File: src/test/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilderTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class StopPlaceQueryFromSearchBuilderTest {
+
+    @Test
+    public void handleCommonWordsInQuery() {
+
+        StopPlaceQueryFromSearchBuilder stopPlaceQueryFromSearchBuilder = new StopPlaceQueryFromSearchBuilder();
+        stopPlaceQueryFromSearchBuilder.commonWordsToIgnore.add("des");
```
Comment (csolem) [inline src/test/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilderTest.java]: Maybe it's better to use constructor argument for common words to ignore? So that the class is more or less immutable?

File: src/test/resources/application.properties [hunk 5]
```diff
@@ -58,4 +58,6 @@ security.ignored=/**
 
 netex.import.enabled.types=MERGE,INITIAL,ID_MATCH,MATCH
 netexXmlReferenceValidator.throwOnValidationError=true
-asyncNetexExport.validateAgainstSchema=true
\ No newline at end of file
+asyncNetexExport.validateAgainstSchema=true
+
+stopPlaces.search.commonWordsToIgnore=des,de,les,le,la,du
```
Comment (csolem) [inline src/test/resources/application.properties]: Can this be removed, as it is probably only relevant for french?

File: src/test/java/org/rutebanken/tiamat/rest/netex/publicationdelivery/ImportResourceTest.java [hunk 6]
```diff
@@ -1003,14 +1020,14 @@ public void importStopPlaceWithMultipleValidBetweenPeriodsIgnoresAllButFirst() t
         LocalDateTime firstValidFrom = LocalDateTime.now().plusSeconds(1);
         LocalDateTime secondValidFrom = LocalDateTime.now().plusSeconds(2);
         StopPlace stopPlace1 = new StopPlace()
-                                       .withId("XYZ:Stopplace:1")
```
Comment (csolem) [inline src/test/java/org/rutebanken/tiamat/rest/netex/publicationdelivery/ImportResourceTest.java]: Formatting changes can optionally be posted in separate PR for clarity.

File: src/main/java/org/rutebanken/tiamat/exporter/PublicationDeliveryExporter.java [hunk 7]
```diff
@@ -91,7 +91,7 @@ public PublicationDeliveryStructurePage exportStopPlacesWithEffectiveChangeInPer
     public PublicationDeliveryStructure createPublicationDelivery() {
         PublicationDeliveryStructure publicationDeliveryStructure = new PublicationDeliveryStructure()
                 .withVersion(ANY_VERSION)
-                .withPublicationTimestamp(OffsetDateTime.now())
+                .withPublicationTimestamp(LocalDateTime.now())
```
Comment (csolem) [inline src/main/java/org/rutebanken/tiamat/exporter/PublicationDeliveryExporter.java]: This has been resolved in our master. Please update your fork from Entur master. One tip is to create a new branch from Entur master for a new PR. Then merge it to your fork's master, and create PR to us :+1:

File: pom.xml [hunk 8]
```diff
@@ -408,7 +408,7 @@
                 https://github.com/jboss/xerces/pull/14
                 https://issues.apache.org/jira/browse/XERCESJ-1276
             -->
-            <version>2.11.0.SP6-RB</version>
+            <version>2.11.0.SP6-RB-SNAPSHOT</version>
```
Comment (csolem) [inline pom.xml]: This dependency is unfortunately only on our nexus. Not sure if adding *-SNAPSHOT* helps. But it might be possible for you to remove the dependency (slightly impacts performance of xml xsd validation). Or we can provide the jar for you. Anyways, this is not relevant to the PR.

File: src/main/java/org/rutebanken/tiamat/netex/id/ValidPrefixList.java [hunk 9]
```diff
@@ -15,27 +15,30 @@
...
+import java.util.List;
+import java.util.Map;
 
 @Component
 public class ValidPrefixList {
 
     public static final String ANY_PREFIX = "*";
 
+    @Value("${netex.validPrefix}")
+    public static final String VALID_NETEX_PREFIX = "NSR";
```
[the remark this answers is not in the document]
Comment (mhicauber) [reply src/main/java/org/rutebanken/tiamat/netex/id/ValidPrefixList.java]: Can't be final then. Not a problem ?

File: src/test/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilderTest.java (hunk 4 above)
Comment (mhicauber) [reply src/test/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilderTest.java]: like this ?

```
/**
     * Configure some common words to be skipped during stop place search by name.
     */
    private Set<String> commonWordsToIgnore;

    public StopPlaceQueryFromSearchBuilder(@Value("${stopPlaces.search.commonWordsToIgnore}") Set<String> commonWordsToIgnore) {
        this.commonWordsToIgnore = commonWordsToIgnore;
    }
```

File: src/main/java/org/rutebanken/tiamat/netex/id/ValidPrefixList.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (csolem) [reply src/main/java/org/rutebanken/tiamat/netex/id/ValidPrefixList.java]: Not a problem.

File: src/test/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilderTest.java (hunk 4 above)
Comment (csolem) [reply src/test/java/org/rutebanken/tiamat/repository/search/StopPlaceQueryFromSearchBuilderTest.java]: Yes. Very nice. Does it handle empty set?

**Comment by mhicauber** [comment]
Fixes as requested, feel free to comment.
Also fixed word skipping functionnality  and added tests/

**Comment by csolem** [comment]
Thanks!
